content/doc/pipeline/index.adoc

```diff
@@ -0,0 +1,981 @@
+   }
+   stash excludes: 'target/', includes: '**', name: 'source'
+}
+stage 'test'
+parallel 'integration': {
+   node {
+      unstash 'source'
+      withEnv(["PATH+MAVEN=${tool 'm3'}/bin"]) {
+         sh "mvn clean verify"
+      }
+   }
+}, 'quality': {
+   node {
+      unstash 'source'
+      withEnv(["PATH+MAVEN=${tool 'm3'}/bin"]) {
+         sh "mvn sonar:sonar"
+      }
+   }
+}
+stage 'approve'
+timeout(time: 7, unit: 'DAYS') {
+   input message: 'Do you want to deploy?', submitter: 'ops'
+}
+stage name:'deploy', concurrency: 1
+node {
+   unstash 'source'
+   withEnv(["PATH+MAVEN=${tool 'm3'}/bin"]) {
+     sh "mvn cargo:deploy"
+   }
+}
+```
+
+# Docker with Pipeline
+
+The Docker Pipeline plugin exposes a docker global variable that provides DSL for common Docker operations, only requiring a Docker client on the executor running the steps (use a label in your node step to target a Docker-enabled slave).
+
+By default, the docker global variable connects to the local Docker daemon. You may use the docker.withServer step to connect to a remote Docker host. The image step provides a handle to a specific Docker image and allows executing several other image related steps, including the image.inside step.  The inside step will start up the specified container and run a block of steps in that container:
+
+```groovy
+docker.image('maven:3.3.3-jdk8').inside('-v ~/.m2/repo:/m2repo') {
+   sh 'mvn -Dmaven.repo.local=/m2repo clean package'
+}
+```
+
+When the steps are complete, the container will be stopped and removed. There are many more features of the Docker Pipeline plugin; additional steps are outlined in the Full Syntax Refcard.
+
+# Extending Pipeline
+
+Like all Jenkins features, Pipeline relies on Jenkins' extensible architecture, allowing developers to extend Pipeline's features.
+
+## Plugin Compatibility
+
+There are a large number of existing plugins for Jenkins. Many of these plugins integrate with Pipeline as build steps, wrappers, and so on. Plugin maintainers must ensure their plugins are Pipeline-compatible. The community has documented the steps to ensure compatibility. More details on plugin development and Pipeline compatibility are on the jenkins-ci.org Wiki: https://wiki.jenkins-ci.org/display/JENKINS/Plugin+tutorial
+
+## Custom DSL
+
+Beyond compatibility, plugin maintainers can also add specific Pipeline DSL for their plugins' behavior. The community has documented the steps to take to add plugin-specific DSL. Examples include the Credentials Binding Plugin, which contributes the withCredentials syntax.
+
+# Full Syntax Reference Card
+
+Following is a full Jenkins Pipeline syntax reference card. Of course, as you add plugins or as plugins are updated new Pipeline Script elements will become available in your environment. The Pipeline snippet generator and UI will automatically add these and any associated help text so you know how to use them!
+
+## Basics
+
+<table>
```

Comment:
All this HTML table mess doesn't render properly. Asciidoc does [support rendering tables](http://asciidoctor.org/docs/user-manual/#tables) though.